### PR TITLE
8260668: vectorIntrinsics windows build problem

### PR DESCRIPTION
**Pull request by sviswa7**
The panama-vector vectorIntrinsics branch fails to build on windows.
This is happening because /Ta option is no more supported by the VS2019 C compiler to generate obj from assembly files.
The alternative is to use the windows assembler instead for this purpose.

Best Regards,
Sandhya

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [JDK-8260668](https://bugs.openjdk.java.net/browse/JDK-8260668): vectorIntrinsics windows build problem

### Reviewers
 * [Magnus Ihse Bursie](https://openjdk.java.net/census#ihse) (@magicus - no project role)
 * [Paul Sandoz](https://openjdk.java.net/census#psandoz) (@PaulSandoz - Committer)

### Download
`$ git fetch https://git.openjdk.java.net/panama-vector pull/34/head:pull/34`
`$ git checkout pull/34`

### CODE REVIEW

**Comment by PaulSandoz** [comment]
@magicus can you please review this?

File: make/autoconf/toolchain.m4 [hunk 1]
```diff
@@ -696,8 +696,13 @@ AC_DEFUN_ONCE([TOOLCHAIN_DETECT_TOOLCHAIN_CORE],
   if test "x$TOOLCHAIN_TYPE" != xmicrosoft; then
     AS="$CC -c"
   else
-    # On windows, the assember is "ml.exe"
-    UTIL_LOOKUP_TOOLCHAIN_PROGS(AS, ml)
+    if test "x$OPENJDK_TARGET_CPU" = "xx86_64"; then
```
Comment (magicus) [inline make/autoconf/toolchain.m4:699]: Is the proper test for x86_64 or for 64-bit CPUs? Or phrased differently, what is the name of the executable on aarch64? The comment below states that the test is for word size, but the test itself is for CPU architecture.

File: src/hotspot/os_cpu/windows_x86/globals_vectorApiSupport_windows.hpp [hunk 2]
```diff
@@ -1,37 +1,37 @@
...
-IF @Version GE 1410
-__VECTOR_API_MATH_INTRINSICS_WINDOWS TEXTEQU <"vector_api">
-ELSE
-__VECTOR_API_MATH_INTRINSICS_WINDOWS TEXTEQU <>
-ENDIF
-
-ELSE
-__VECTOR_API_MATH_INTRINSICS_WINDOWS TEXTEQU <>
-ENDIF
+; Copyright (c) 1997, 2021, Oracle and/or its affiliates. All rights reserved.
```
Comment (magicus) [inline src/hotspot/os_cpu/windows_x86/globals_vectorApiSupport_windows.hpp:1]: Maybe you have already established this as a new convention, but I think it is highly confusing to have a file named `.hpp` which is not a C++ header file! I expect IDEs etc will get confused too, and complain about syntax errors. Recommend renaming it to something different. If no conventional suffix exists for assembly language includes, perhaps just name it `.s.inc`?

File: make/common/NativeCompilation.gmk [hunk 3]
```diff
@@ -438,7 +438,8 @@ define SetupCompileNativeFileBody
             # For assembler calls, no need to build dependency list.
 	    $$(call ExecuteWithLog, $$@, $$(call MakeCommandRelative, \
 	        $$($1_COMPILER) $$($1_FLAGS) \
-	          $(CC_OUT_OPTION)$$($1_OBJ) /Ta $$($1_SRC_FILE)))
+	          $(CC_OUT_OPTION)$$($1_OBJ) /nologo /c /Ta $$($1_SRC_FILE) 2>&1 \
+	            | $(TR) -d '\r' | $(GREP) -v -e "Assembling:" || test "$$$$?" = "1" ;))
```
Comment (magicus) [inline make/common/NativeCompilation.gmk:442]: Please use the `-nologo -c -Ta` format for the options.

**Review by magicus — CHANGES_REQUESTED** [review]

File: make/autoconf/toolchain.m4 (hunk 1 above)
Comment (sviswa7) [reply make/autoconf/toolchain.m4]: Changed to:
   if test "x$OPENJDK_TARGET_CPU_BITS" = "x64"; then

File: src/hotspot/os_cpu/windows_x86/globals_vectorApiSupport_windows.hpp (hunk 2 above)
Comment (sviswa7) [reply src/hotspot/os_cpu/windows_x86/globals_vectorApiSupport_windows.hpp:1]: Renamed to .s.inc

File: make/common/NativeCompilation.gmk (hunk 3 above)
Comment (sviswa7) [reply make/common/NativeCompilation.gmk:442]: done

**Comment by sviswa7** [comment]
@magicus Thanks a lot for your review. I have incorporated all your comments. Let me know if I need to change something else.

Best Regards,
Sandhya

**Review by magicus — APPROVED** [review]
Great! Looks good to me now.

**Comment by sviswa7** [comment]
@PaulSandoz Please approve this PR if it looks ok to you.

**Review by PaulSandoz — APPROVED** [review]
Looks ok, but i am not familiar with this area, so piggy backing off @magicus approval.

**Pull request closed without merging**